### ISSUE TITLE
Bug in nni/compression/pytorch/speedup/infer_shape.py, line no. 625-626.

### ISSUE THREAD

**Issue by Davidxswang** [microsoft/nni#3562]
Line no. 595 - 630, in  nni/compression/pytorch/speedup/infer_shape.py, is the function `view_outshape`, which is used to infer the input shape (B C H W) from the output shape (B L), as shown below. `mask.mask_index[1]` is the mask on the `L` dimension of the output tensor. Based on the context, I think what the line 625-626 should do is to find out the mask for the `C` dimension of the input tensor.

By the way, the function description seems wrong as the `mask` should be `The mask of its output tensor`, and the `Returns` should be `The mask of its input tensor`.

```
def view_inshape(module_masks, mask, shape):                                                 # line no. 547
    """
    This is a limited support

    TODO: consider replace tensor.view with nn.Flatten, because tensor.view is not
    included in module, thus, cannot be replaced by our framework.

    Parameters
    ----------
    module_masks : ModuleMasks
        The ModuleMasks instance of the ```view``` op
    mask : CoarseMask
        The mask of its input tensor
    shape : dict
        Original shape of its input and output tensors

    Returns
    -------
    CoarseMask
        The mask of its output tensor
    """
    # NOTE: the case constrained by the following four asserts
    assert shape['in_shape'][0] == shape['out_shape'][0]
    assert len(shape['in_shape']) == 4
    assert len(shape['out_shape']) == 2
    assert shape['out_shape'][1] == shape['in_shape'][1] * \
        shape['in_shape'][2]*shape['in_shape'][3]

    assert isinstance(mask, CoarseMask)
    assert mask.mask_index[1] is not None
    assert mask.mask_index[0] is None
    assert mask.mask_index[2] is None
    assert mask.mask_index[3] is None
    # due to the cat operation, the same node may be
    # accessed more than once
    if module_masks.input_mask is not None:
        assert module_masks.input_mask <= mask
    module_masks.set_input_mask(mask)
    output_cmask = CoarseMask(num_dim=2)
    index = []
    step_size = shape['in_shape'][2] * shape['in_shape'][3]
    for loc in mask.mask_index[1]:
        index.extend([loc * step_size + i for i in range(step_size)])
    output_cmask.add_index_mask(dim=1, index=torch.tensor(index).to(mask.mask_index[1].device))  # pylint: disable=not-callable
    module_masks.set_output_mask(output_cmask)
    return output_cmask


def view_outshape(module_masks, mask, shape):                                       # line no. 595
    """
    Parameters
    ----------
    module_masks : ModuleMasks
        The ModuleMasks instance of the ```flatten``` op
    mask : CoarseMask
        The mask of its input tensor
    shape : dict
        Original shape of its input and output tensors
    Returns
    -------
    CoarseMask
        The mask of its output tensor
    """
    # NOTE: the case constrained by the following four asserts
    assert shape['in_shape'][0] == shape['out_shape'][0]
    assert len(shape['in_shape']) == 4
    assert len(shape['out_shape']) == 2
    assert shape['out_shape'][1] == shape['in_shape'][1] * \
        shape['in_shape'][2]*shape['in_shape'][3]

    assert isinstance(mask, CoarseMask)
    assert mask.mask_index[1] is not None
    assert mask.mask_index[0] is None

    module_masks.set_output_mask(mask)
    input_cmask = CoarseMask(num_dim=4)
    index = []
    step_size = shape['in_shape'][2] * shape['in_shape'][3]
    for loc in mask.mask_index[1]:                                                        # line no. 625
        index.extend([loc * step_size + i for i in range(step_size)])
    input_cmask.add_index_mask(dim=1, index=torch.tensor(index).to(mask.mask_index[1].device))  # pylint: disable=not-callable
    module_masks.set_input_mask(input_cmask)

    return input_cmask
```

**Comment by J-shang** [comment]
Hi @Davidxswang , thanks for your contribution, yes, it is a bug, maybe this can fix it. If it is convenient, you can make a pr to fix it.

```
index = set()
step_size = shape['in_shape'][2] * shape['in_shape'][3]
for loc in mask.mask_index[1]:                                                        # line no. 625
    index.add(loc // step_size)
index = list(index)
```

**Comment by Davidxswang** [comment]
> Hi @Davidxswang , thanks for your contribution, yes, it is a bug, maybe this can fix it. If it is convenient, you can make a pr to fix it.
> 
> ```
> index = set()
> step_size = shape['in_shape'][2] * shape['in_shape'][3]
> for loc in mask.mask_index[1]:                                                        # line no. 625
>     index.add(loc // step_size)
> index = list(index)
> ```

Thank you! I have submitted the PR.

**Comment by Davidxswang** [comment]
I'll close this issue since it has been solved and merged into master branch.
